### ISSUE TITLE
ID mapper issue

### ISSUE THREAD

**Issue by katewarner** [glygener/glygen-issues#1870]
I think I worked out the problem with the ID mapper.

It looks like the ID mapper currently works like this: 
From ID ---> Internal anchor ID (UniProtKB Canonical Ac) ---> To ID: Canonical protein and all mapped isoforms (whether or not they contain the "From ID")

This can be incorrect because the xref ("From ID") is not always present in the unreviewed isoform entries. For example, the screenshot below contains the results for mapping GeneCards IDs (using the default example) to UniProt IDs: https://www.glygen.org/mapper-result/4be4d5683d0541c382642f75a9fe75dd

<img src="https://github.com/user-attachments/assets/bf4cadb2-709b-429a-9f05-e82566878093" width="400"/>

So the GeneCards ID "ENO3" maps to the canonical protein "P13929" and all the unreviewed isoforms mapped to the canonical protein. In UniProt "ENO3" only maps to "P13929", and this is also the case in our human EBI NT file.

```
[k.warner1@glygen-vm-dev 2024_06_20]$ grep 'genecards/ENO3' uniprot-proteome-homo-sapiens.nt
<http://purl.uniprot.org/uniprot/P13929> <http://www.w3.org/2000/01/rdf-schema#seeAlso> <http://purl.uniprot.org/genecards/ENO3> .
<http://purl.uniprot.org/genecards/ENO3> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://purl.uniprot.org/core/Resource> .
<http://purl.uniprot.org/genecards/ENO3> <http://purl.uniprot.org/core/database> <http://purl.uniprot.org/database/GeneCards> .
``` 

Some IDs do map to the canonical protein and all the unreviewed isoforms, so the ID mapper should probably work like this:
From ID ---> Internal anchor ID (UniProtKB Canonical Ac) ---> To ID: Any protein mapped to the "From ID"

**Comment by jeet-vora** [comment]
@katewarner 
Seems there is another issue
I was also referring to Genecards to AlphafoldDB. It does not seem to be correct.

We can talk more when we meet next.

**Comment by katewarner** [comment]
@jeet-vora Yes sorry, I forgot to add that bit. The AlphaFold ID problem is that a GeneCard ID will map to the canonical protein and the unreviewed isoforms (as in the example above), but the canonical protein and each of the unreviewed isoforms have a unique AlphaFold ID. 

<img src="https://github.com/user-attachments/assets/46ef26b0-8e52-4578-924b-5f7ff9d95f22" width="400"/>

This should be fixed at the next release because for the 2.7 AlphaFold datasets Robel is only mapping to the canonical protein, whereas in the last release he was mapping to the canonical proteins and the unreviewed proteins.

**Comment by katewarner** [comment]
@rykahsay Add drop-down – “UniProtKB canonical only” or “all isoforms”

**Comment by rykahsay** [comment]
This is not clear for me ... please bring this up in our next meeting

**Comment by katewarner** [comment]
Bring up at Monday meeting
